src/main/java/com/pnu/dev/radioserviceapi/controller/admin/ScheduleAdminController.java

```diff
@@ -1,12 +1,202 @@
+                .orElseThrow(() -> new RadioServiceAdminException("Не існуючий день тижня"))
+                .toDayOfWeekResponse();
+
+        model.addAttribute("dayOfWeek", dayOfWeek);
+
+        return "schedule/newItem";
+    }
+
+    @GetMapping("/item/delete/{id}")
+    public String deleteConfirmation(Model model, @PathVariable("id") String id, HttpServletRequest request) {
+
+        findByIdOrThrowException(id);
+
+        model.addAttribute("message", "Ви впевнені, що справді хочете запис з розкладу?");
+        model.addAttribute("returnBackUrl", HttpUtils.getPreviousPageUrl(request));
+
+        return "common/deleteConfirmation";
+    }
+
+    @PostMapping("/item/new")
+    public String create(@ModelAttribute NewScheduleItemForm newScheduleItemForm,
+                         RedirectAttributes redirectAttributes,
+                         HttpServletRequest httpServletRequest) {
+
+        OperationResult<ScheduleItemDto> scheduleItemOperationResult = scheduleItemService.create(newScheduleItemForm);
+
+        if (scheduleItemOperationResult.isError()) {
+            redirectAttributes.addFlashAttribute(FLASH_ERROR, scheduleItemOperationResult.getErrorMessage());
+            return redirectToPreviousPage(httpServletRequest);
+        }
+
+        redirectAttributes.addFlashAttribute(FLASH_MESSAGE, "Запис було успішно створено");
+
+        ScheduleItemDto scheduleItem = scheduleItemOperationResult.getData();
+        return String.format("redirect:/admin/schedule/day/%s?selectedItemId=%s",
+                newScheduleItemForm.getDayOfWeekUrlValue(), scheduleItem.getId());
+    }
+
+    @PostMapping("/item/update/{id}")
+    public String update(@PathVariable("id") String id,
+                         @ModelAttribute UpdateScheduleItemForm updateScheduleItemForm,
+                         RedirectAttributes redirectAttributes,
+                         HttpServletRequest httpServletRequest) {
+
+        OperationResult<ScheduleItemDto> scheduleItemOperationResult = scheduleItemService
+                .update(id, updateScheduleItemForm);
+
+        if (scheduleItemOperationResult.isError()) {
+            redirectAttributes.addFlashAttribute(FLASH_ERROR, scheduleItemOperationResult.getErrorMessage());
+            return redirectToPreviousPage(httpServletRequest, id);
+        }
+
+        redirectAttributes.addFlashAttribute(FLASH_MESSAGE, "Запис було успішно оновлено");
+
+        ScheduleItemDto scheduleItem = scheduleItemOperationResult.getData();
+        return String.format("redirect:/admin/schedule/day/%s?selectedItemId=%s",
+                scheduleItem.getDayOfWeek().getUrlValue(), scheduleItem.getId());
+    }
+
+    @PostMapping("/item/delete/{id}")
+    public String delete(@PathVariable("id") String id, RedirectAttributes redirectAttributes) {
+
+        ScheduleItemDto scheduleItem = findByIdOrThrowException(id);
+
+        scheduleItemService.deleteById(id);
```

Comment:
Why do not check ifExists? in Service layer?

Reply:
In this concrete case we use value of `DayOfWeek` from the `findByIdMethod` to be able to redirect user to correct page after completing